### DISCUSSION TITLE
Update example with multiple middlewares (next-auth)

### DISCUSSION THREAD

**Discussion by RomkaLTU** [amannn/next-intl#697] [Q&A]
### Is your feature request related to a problem? Please describe.

The problem with the current example with next-auth middleware is that it's not really a use case in the real-world app, unless you just need some admin zone without translations.

The main problem is here:

```
if (isProtectedPage) {
    return (authMiddleware as any)(req);
  } else {
    return intlMiddleware(req);
  }
```

So you need to choose between 2 middlewares, but usually, you need two at the same time. 

### Describe the solution you'd like

I would like to have translations inside withAuth() middleware, some combination of 2. 

### Describe alternatives you've considered

Tried example: https://github.com/amannn/next-intl/tree/main/examples/example-app-router-next-auth

**Comment by amannn** [comment]
The NextAuth.js example uses i18n for both public as well as protected pages, you might be interested in the corresponding docs here: [Example: Integrating with Auth.js (aka NextAuth.js)](https://next-intl-docs.vercel.app/docs/routing/middleware#example-auth-js).

I'll move this to a discussion since it seems to be a usage question.
